crates/agglayer/src/main.rs

```diff
@@ -31,6 +34,51 @@ fn main() -> anyhow::Result<()> {
             let vkey = agglayer_prover::get_vkey();
             println!("{}", vkey);
         }
+
+        cli::Commands::Backup(cli::Backup::List { config_path: cfg }) => {
+            let cfg = agglayer_config::Config::try_load(&cfg)?;
+
+            if let BackupConfig::Enabled { path, .. } = cfg.storage.backup {
+                match agglayer_storage::storage::backup::BackupEngine::list_backups(&path) {
+                    Ok(result) => println!("{}", serde_json::to_string(&result).unwrap()),
+                    Err(error) => eprintln!("{}", error),
+                }
+            }
+        }
+
+        cli::Commands::Backup(cli::Backup::Restore {
+            config_path: cfg,
+            db_versions,
+        }) => {
+            let cfg = agglayer_config::Config::try_load(&cfg)?;
+
+            if let BackupConfig::Enabled { path, .. } = cfg.storage.backup {
+                for (db_kind, version) in db_versions {
+                    let (db_path, backup_path) = match db_kind {
+                        cli::DbKind::State => {
+                            (cfg.storage.state_db_path.join("state"), path.join("state"))
+                        }
+                        cli::DbKind::Pending => (
+                            cfg.storage.pending_db_path.join("pending"),
+                            path.join("pending"),
+                        ),
+                        cli::DbKind::Epoch(epoch_number) => (
+                            cfg.storage.epochs_db_path.join(format!("{}", epoch_number)),
+                            path.join(format!("epochs/{}", epoch_number)),
+                        ),
+                    };
```

Comment:
Nit: The path calculation could be a separate function. I suspect it's also useful where the backups are generated?